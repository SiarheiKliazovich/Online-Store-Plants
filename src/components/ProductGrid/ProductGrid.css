.maim__product_wrapper {
  display: flex;
  flex-wrap: wrap;
}

.container-inner {
  padding: 15px;
  width: 33%;
}

.product-block-inner {
  border-radius: 5px;
  background: #ffffff;
  border: 2px solid #e4e4e4;
  overflow: hidden;
  text-align: center;
}

.product__img_link {
  color: #000;
}

.product__img {
  width: 100%;
}

.product__text {
  color: #000;
  font-size: 16px;
  font-family: Work Sans, sans-serif;
  font-weight: 500;
  letter-spacing: 0.2px;
  padding: 5px 5px;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
}

.price {
  padding-bottom: 10px;
}

.product__quantity {
  display: flex;
  flex-direction: column;
  margin-top: 5px;
  margin-bottom: 5px;
}

.product__input_wrapper {
  display: flex;
  flex-direction: row;
  justify-content: center;
}

.product__quantity_title {
  font-family: Work Sans, sans-serif;
  font-size: 18px;
  padding-right: 5px;
}

.product__quantity_input-text {
  width: 30px;
  text-align: center;
  margin-left: 5px;
  margin-right: 5px;
}

.product__btn {
  margin: 10px 0;
  padding: 8px 15px 7px;
  height: auto;
  background-color: #014e37;
  color: #ffffff;
  border-color: #014e37;
  font-family: Work Sans, sans-serif;
  border-radius: 5px;
  cursor: pointer;
  text-decoration: none;
  text-transform: uppercase;
}

.product__btn:hover {
  background-color: #05946a;
}

/*# sourceMappingURL=ProductGrid.css.map */
