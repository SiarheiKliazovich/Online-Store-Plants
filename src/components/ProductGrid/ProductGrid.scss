@import "../../sass/variables";

.maim__product_wrapper {
  display: flex;
  flex-wrap: wrap;
}

.container-inner {
  padding: 15px;
  width: 33%;
}

.product-block-inner{
  border-radius: 5px;
  background: $color-white;
  border: 2px solid $color-grey;
  overflow: hidden;
  text-align: center;
}

.product__img_link {
  color: $color-black;
}

.product__img{
  width: 100%;
}

.product__text {
  color: $color-black;
  font-size: $font-size-md;
  font-family: $font-family-primary;
  font-weight: $font-weight-bold;
  letter-spacing: 0.2px;
  padding:5px 5px;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
}

.price{
  padding-bottom: 10px;
}



.product__quantity {
  display: flex;
  flex-direction: column;
  margin-top: 5px;
  margin-bottom: 5px;
}

.product__input_wrapper {
  display: flex;
  flex-direction: row;
  justify-content: center;
}


.product__quantity_title {
  font-family: $font-family-primary;
  font-size: $font-size-lg;
  padding-right: 5px;
}

.product__quantity_input-text {
  width: 30px;
  text-align: center;
  margin-left: 5px;
  margin-right: 5px;
  
}

.product__btn {
  margin: 10px 0;
  padding: 8px 15px 7px;
  height: auto;
  background-color:$color-green-black;
  color: $color-white;
  border-color: $color-green-white;
  font-family: Work Sans, sans-serif;
  border-radius: 5px;
  cursor: pointer;
  text-decoration: none;
  text-transform: uppercase;
}

.product__btn:hover {
  background-color:  $color-green-hover;
}