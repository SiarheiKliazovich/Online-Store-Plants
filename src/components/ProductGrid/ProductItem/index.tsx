import { FunctionComponent } from "react";
import { Product } from "../../../interface/product";
import { Link } from "react-router-dom";
import "./ProductItem.scss";

type ProductType = {
  product: Product;
};

const ProductItem: FunctionComponent<ProductType> = ({
  product,
}: ProductType) => {
  return (
    <div className="container-inner" key={product.id}>
      <div className="product-block-inner">
        <Link className="product__img_link" to={`/product/${product.id}`}>
          <img
            className="product__img"
            src={product.thumbnail}
            alt="image product"
          />
          <div className="product__text_wrapper">
            <h3 className="product__text">{product.name}</h3>
            <span className="price">
              ${product.price[0]} - ${product.price[product.price.length - 1]}
            </span>
          </div>
        </Link>
        <div className="product__quantity">
          <div className="product__input_wrapper">
            <span className="product__quantity_title">Qty: </span>
            <input className="minus" type="button" value="-" />
            <input
              className="product__quantity_input-text"
              type="number"
              step="1"
              min="1"
              max="1000"
              name="quantity"
              value="1"
              title="Qty"
              pattern="[0-9]*"
            />
            <input className="plus" type="button" value="+" />
          </div>
          <div className="product__btn_wrapper">
            <button className="product__btn">Add to cart</button>
          </div>
        </div>
      </div>
    </div>
  );
};

export default ProductItem;
