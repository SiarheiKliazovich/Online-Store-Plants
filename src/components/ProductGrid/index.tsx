import ProductItem from "./ProductItem";
import "./ProductGrid.scss";
import classNames from "classnames";
import { FunctionComponent } from "react";
import { ProductGridType } from "../../types";

const ProductGrid: FunctionComponent<ProductGridType> = ({
  view,
  products,
  shoppingCart,
  addToShoppingCart,
}: ProductGridType) => (
  <div
    className={classNames("main__product_wrapper", {
      line: view === "line",
    })}
  >
    {products.map((product) => (
      <ProductItem
        product={product}
        key={product.id.toString()}
        shoppingCart={shoppingCart}
        addToShoppingCart={addToShoppingCart}
      />
    ))}
  </div>
);

export default ProductGrid;
