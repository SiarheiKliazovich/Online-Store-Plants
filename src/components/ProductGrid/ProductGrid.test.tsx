import { render } from "@testing-library/react";
import { MemoryRouter as Router } from "react-router-dom";
import ProductGrid from "../ProductGrid";

it("ProductGrid component when view is grid", () => {
  const view = "grid";
  const products = [
    {
      id: 1,
      name: "Biotech",
      title: "The growth of a plant involves several stages.",
      description: ["Fast, a plot of land is selected."],
      size: "XL",
      color: "Pink",
      availability: "In Stock",
      price: 25,
      category: "Aeonium",
      stock: 3,
      rating: 4,
      brand: "London-Plants&Co",
      thumbnail:
        "https://envatopixel.com/wordpress/plantlish/wp-content/uploads/2021/07/2.webp",
      images: [
        "https://envatopixel.com/wordpress/plantlish/wp-content/uploads/2021/07/2.webp",
      ],
    },
  ];
  const shoppingCart = [{ id: 1, count: 1 }];
  const addToShoppingCart = jest.fn();
  const wrapper = render(
    <Router>
      <ProductGrid
        view={view}
        products={products}
        shoppingCart={shoppingCart}
        addToShoppingCart={addToShoppingCart}
      />
    </Router>
  );
  expect(wrapper).toMatchSnapshot();
});

it("ProductGrid component when view is line", () => {
  const view = "line";
  const products = [
    {
      id: 1,
      name: "Biotech",
      title: "The growth of a plant involves several stages.",
      description: ["Fast, a plot of land is selected."],
      size: "XL",
      color: "Pink",
      availability: "In Stock",
      price: 25,
      category: "Aeonium",
      stock: 3,
      rating: 4,
      brand: "London-Plants&Co",
      thumbnail:
        "https://envatopixel.com/wordpress/plantlish/wp-content/uploads/2021/07/2.webp",
      images: [
        "https://envatopixel.com/wordpress/plantlish/wp-content/uploads/2021/07/2.webp",
      ],
    },
  ];
  const shoppingCart = [{ id: 1, count: 1 }];
  const addToShoppingCart = jest.fn();
  const wrapper = render(
    <Router>
      <ProductGrid
        view={view}
        products={products}
        shoppingCart={shoppingCart}
        addToShoppingCart={addToShoppingCart}
      />
    </Router>
  );
  expect(wrapper).toMatchSnapshot();
});
