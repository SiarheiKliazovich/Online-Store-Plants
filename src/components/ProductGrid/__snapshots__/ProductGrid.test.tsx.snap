// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductGrid component when view is grid 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="main__product_wrapper"
      >
        <div
          class="container-inner"
        >
          <div
            class="product-block-inner product-active"
          >
            <div
              class="img-prod_wrapper"
            >
              <a
                class="product__img_link"
                href="/product/1"
              >
                <img
                  alt="image product"
                  class="product__img"
                  src="https://envatopixel.com/wordpress/plantlish/wp-content/uploads/2021/07/2.webp"
                />
              </a>
            </div>
            <div
              class="product__info_wrapper"
            >
              <a
                class="product__img_link"
                href="/product/1"
              >
                <div
                  class="product__text_wrapper"
                >
                  <h3
                    class="product__text-name"
                  >
                    Biotech
                  </h3>
                  <h4
                    class="product__description_card"
                  >
                    Fast, a plot of land is selected.
                  </h4>
                  <h4
                    class="product__brand_card"
                  >
                    Brand: 
                    London-Plants&Co
                  </h4>
                  <h4
                    class="product__category_card"
                  >
                    Category: 
                    Aeonium
                  </h4>
                  <h4
                    class="product__rating_card"
                  >
                    Rating: 
                    4
                  </h4>
                  <h4
                    class="product__stock_card"
                  >
                    In Stock: 
                    3
                  </h4>
                  <h4
                    class="product__price_card"
                  >
                    Price: $
                    25
                  </h4>
                </div>
              </a>
              <div
                class="product__quantity"
              >
                <div
                  class="product__btn_wrapper"
                >
                  <button
                    class="product__btn"
                  >
                    Drop from cart
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="main__product_wrapper"
    >
      <div
        class="container-inner"
      >
        <div
          class="product-block-inner product-active"
        >
          <div
            class="img-prod_wrapper"
          >
            <a
              class="product__img_link"
              href="/product/1"
            >
              <img
                alt="image product"
                class="product__img"
                src="https://envatopixel.com/wordpress/plantlish/wp-content/uploads/2021/07/2.webp"
              />
            </a>
          </div>
          <div
            class="product__info_wrapper"
          >
            <a
              class="product__img_link"
              href="/product/1"
            >
              <div
                class="product__text_wrapper"
              >
                <h3
                  class="product__text-name"
                >
                  Biotech
                </h3>
                <h4
                  class="product__description_card"
                >
                  Fast, a plot of land is selected.
                </h4>
                <h4
                  class="product__brand_card"
                >
                  Brand: 
                  London-Plants&Co
                </h4>
                <h4
                  class="product__category_card"
                >
                  Category: 
                  Aeonium
                </h4>
                <h4
                  class="product__rating_card"
                >
                  Rating: 
                  4
                </h4>
                <h4
                  class="product__stock_card"
                >
                  In Stock: 
                  3
                </h4>
                <h4
                  class="product__price_card"
                >
                  Price: $
                  25
                </h4>
              </div>
            </a>
            <div
              class="product__quantity"
            >
              <div
                class="product__btn_wrapper"
              >
                <button
                  class="product__btn"
                >
                  Drop from cart
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ProductGrid component when view is line 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="main__product_wrapper line"
      >
        <div
          class="container-inner"
        >
          <div
            class="product-block-inner product-active"
          >
            <div
              class="img-prod_wrapper"
            >
              <a
                class="product__img_link"
                href="/product/1"
              >
                <img
                  alt="image product"
                  class="product__img"
                  src="https://envatopixel.com/wordpress/plantlish/wp-content/uploads/2021/07/2.webp"
                />
              </a>
            </div>
            <div
              class="product__info_wrapper"
            >
              <a
                class="product__img_link"
                href="/product/1"
              >
                <div
                  class="product__text_wrapper"
                >
                  <h3
                    class="product__text-name"
                  >
                    Biotech
                  </h3>
                  <h4
                    class="product__description_card"
                  >
                    Fast, a plot of land is selected.
                  </h4>
                  <h4
                    class="product__brand_card"
                  >
                    Brand: 
                    London-Plants&Co
                  </h4>
                  <h4
                    class="product__category_card"
                  >
                    Category: 
                    Aeonium
                  </h4>
                  <h4
                    class="product__rating_card"
                  >
                    Rating: 
                    4
                  </h4>
                  <h4
                    class="product__stock_card"
                  >
                    In Stock: 
                    3
                  </h4>
                  <h4
                    class="product__price_card"
                  >
                    Price: $
                    25
                  </h4>
                </div>
              </a>
              <div
                class="product__quantity"
              >
                <div
                  class="product__btn_wrapper"
                >
                  <button
                    class="product__btn"
                  >
                    Drop from cart
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="main__product_wrapper line"
    >
      <div
        class="container-inner"
      >
        <div
          class="product-block-inner product-active"
        >
          <div
            class="img-prod_wrapper"
          >
            <a
              class="product__img_link"
              href="/product/1"
            >
              <img
                alt="image product"
                class="product__img"
                src="https://envatopixel.com/wordpress/plantlish/wp-content/uploads/2021/07/2.webp"
              />
            </a>
          </div>
          <div
            class="product__info_wrapper"
          >
            <a
              class="product__img_link"
              href="/product/1"
            >
              <div
                class="product__text_wrapper"
              >
                <h3
                  class="product__text-name"
                >
                  Biotech
                </h3>
                <h4
                  class="product__description_card"
                >
                  Fast, a plot of land is selected.
                </h4>
                <h4
                  class="product__brand_card"
                >
                  Brand: 
                  London-Plants&Co
                </h4>
                <h4
                  class="product__category_card"
                >
                  Category: 
                  Aeonium
                </h4>
                <h4
                  class="product__rating_card"
                >
                  Rating: 
                  4
                </h4>
                <h4
                  class="product__stock_card"
                >
                  In Stock: 
                  3
                </h4>
                <h4
                  class="product__price_card"
                >
                  Price: $
                  25
                </h4>
              </div>
            </a>
            <div
              class="product__quantity"
            >
              <div
                class="product__btn_wrapper"
              >
                <button
                  class="product__btn"
                >
                  Drop from cart
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
